### PR TITLE
Bug fix/do not reclaim shard ownership

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

During a move-shard job which moves the leader there is a
  situation in which the old owner of a shard can reclaim ownership
  (after having resigned already), with a small race where it allows to
  write documents only locally, but then continue the move-shard to a
  server without those documents. An additional bug in the MoveShard
  Supervision job would then leave the shard in a bad configuration
  with a resigned leader permanently in charge.

- [x] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [ ] :book: CHANGELOG entry made
- [ ] :muscle: The behavior in this PR was *manually tested*
- [ ] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [x] Backports required for:
  - [x] 3.7
  - [x] 3.6
  - [x] 3.5

#### Related Information

*(Please reference tickets / specification etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [x] Design document: https://github.com/arangodb/documents/blob/master/DesignDocuments/99_UNSORTED/Supervision.md

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] This change is already covered by existing tests, such as *(please describe tests)*. arangsync, resilience, chaos
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added **Regression Tests**
  - [x] Added new C++ **Unit Tests**
  - [ ] Added new **integration tests** (i.e. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

Additionally:

- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

*(Include link to Jenkins run etc)*

### Documentation

> All new Features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the changelog so that
> developers and users have a concise overview. 

- [ ] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: tests/Maintenance/MaintenanceTest.cpp [hunk 1]
```diff
@@ -715,38 +951,405 @@ TEST_F(MaintenanceTestActionPhaseOne,
   }
 }
 
+
+TEST_F(MaintenanceTestActionPhaseOne, leader_behaviour_plan_self_current_self) {
```
Comment (neunhoef) [inline tests/Maintenance/MaintenanceTest.cpp:955]: In this test suite, `current` is consistently used for `local`, which I find confusing. PhaseOne has nothing to do with Current. We should rename this.

**Review by mchacki — COMMENTED** [review]
The part by @neunhoef LGTM as well.
I am not allowed to APPROVE as I have opened the PR

File: tests/Maintenance/MaintenanceTest.cpp (hunk 1 above)
Comment (mchacki) [reply tests/Maintenance/MaintenanceTest.cpp]: Done!

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by mchacki** [comment]
Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11531/

**Comment by neunhoef** [comment]
LGTM

**Comment by mchacki** [comment]
Jenkins again:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11533/

**Comment by neunhoef** [comment]
https://jenkins01.arangodb.biz/job/arangodb-matrix-pr/11549/

**Pull request merged**